src/detect-engine-threshold.c

```diff
@@ -150,7 +150,7 @@ int ThresholdTimeoutCheck(Host *host, struct timeval *tv)
 
     prev = NULL;
     while (tmp != NULL) {
-        if ((tv->tv_sec - tmp->tv_sec1) <= tmp->seconds) {
+        if (abs(tv->tv_sec - tmp->tv_sec1) <= tmp->seconds) {
```

Comment:
what do you mean "effect on time handling in general"? Is there anything you would like to be modified in a different way?